### ISSUE TITLE
Enhancement: Allow to switch projects while a local job is running

### ISSUE THREAD

**Issue by Timag** [rstudio/rstudio#5163]
**TLDR;**
Switch RStudio projects while a local job is running.

**Introduction:**
If i work on Project A, i open the corresponing Project A environment with Rstudio projects. Then i start a script as a local job. As there is currently nothing to do for project A it would be great to be able to switch to the Project B environment with RStudio projects (and let the local job keep running for Project B).

**Current behaviour:**
Currently I am asked to end the local job before being able to switch projects.

P.S
I am a big fan of the Rstudio Projects and the new Job functionality. I have no clue how technically difficult it is to implement such a change, just wanted to contribute it as an idea.

**Comment by jmcphers** [comment]
The reason it's like this now is that local jobs are children of the R session. So if you end your R session by e.g. restarting or switching projects, you also lose all your jobs. It'd be nicer if we moved these up a level so that they were children of the desktop frame (or some other more persistent supervisor)! This issue tracks that work: https://github.com/rstudio/rstudio/issues/2059
